lib/auth/iam.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+/**
+ * IAM credentials.
+ *
+ * @param {string=} selector the iam authority selector
+ * @param {string=} token the token
+ * @constructor
+ */
+function IAM(selector, token) {
```

Comment:
This name seems a little non specific since IAM is a technology set. Perhaps IAMCredentials or IAMAuth?